shared/src/main/scala/mlscript/ucs/stages/Transformation.scala

```diff
@@ -169,15 +237,17 @@ trait Transformation { self: mlscript.pretyper.Traceable =>
 }
 
 object Transformation {
-  private object OperatorIs {
+  private def rare: Nothing = lastWords("found a very rare case during desugaring UCS terms")
```

Comment:
What does it mean for something to be "rare"? And do you think it looks serious if the compiler tells the user they hit a """rare""" case? Why not just use `die`?

Reply:
So far, this message is for my own reference, because I found some unconsidered transformation cases while I was migrating test cases. However, I can also remove it because I found that I can still view stack trace.